### PR TITLE
check for nullness

### LINKED ISSUE #59: Example of netcdf => xarray => rechunker => zarr?
I saw that there was experimental support for xarray datasets, so I tried this hopefully reproducible notebook that seeks to read from OPeNDAP and write rechunked zarr: 

https://gist.github.com/7f3a4367224aeaee227bdfa90620ed96

The code:
```python
max_mem = '4GB'
array_plan = rechunk(ds, chunk_plan, max_mem, target, temp_store=temp)
```
results in this error:
```python-traceback
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-11-969522ca9241> in <module>
      1 max_mem = '4GB'
----> 2 array_plan = rechunk(ds, chunk_plan, max_mem, target, temp_store=temp)

~/miniconda3/envs/pangeo/lib/python3.7/site-packages/rechunker/api.py in rechunk(source, target_chunks, max_mem, target_store, target_options, temp_store, temp_options, executor)
    301         target_options=target_options,
    302         temp_store=temp_store,
--> 303         temp_options=temp_options,
    304     )
    305     plan = executor.prepare_plan(copy_spec)

~/miniconda3/envs/pangeo/lib/python3.7/site-packages/rechunker/api.py in _setup_rechunk(source, target_chunks, max_mem, target_store, target_options, temp_store, temp_options)
    381                 temp_store_or_group=temp_group,
    382                 temp_options=options,
--> 383                 name=name,
    384             )
    385             copy_spec.write.array.attrs.update(variable_attrs)  # type: ignore

~/miniconda3/envs/pangeo/lib/python3.7/site-packages/rechunker/api.py in _setup_array_rechunk(source_array, target_chunks, max_mem, target_store_or_group, target_options, temp_store_or_group, temp_options, name)
    464 
    465     if isinstance(target_chunks, dict):
--> 466         array_dims = _get_dims_from_zarr_array(source_array)
    467         try:
    468             target_chunks = _shape_dict_to_tuple(array_dims, target_chunks)

~/miniconda3/envs/pangeo/lib/python3.7/site-packages/rechunker/api.py in _get_dims_from_zarr_array(z_array)
    138     # use Xarray convention
    139     # http://xarray.pydata.org/en/stable/internals.html#zarr-encoding-specification
--> 140     return z_array.attrs["_ARRAY_DIMENSIONS"]
    141 
    142 

AttributeError: 'Array' object has no attribute 'attrs'
```

Is it expected that rechunker can be used this way with xarray src data?

If not, does someone have an example of using rechunker to write zarr from xarray src data?

### PR DESCRIPTION
**Pull request by ocefpaf**
This should fix #59. It was failing b/c a `fsspec.get_mapper()` evaluates to `False`. I don't know enough about `fsspec` but I guess that makes sense if a directory is not created yet. Regardless we should always check for nullness instead of `if var: do it` in those cases.

### CODE REVIEW

**Comment by rabernat** [comment]
I'm going to cc @martindurant on this. It seems unexpected that `fsspec.get_mapper()` evaluates to False. Martin, is this expected behavior or an fsspec bug?

**Comment by martindurant** [comment]
Since it derives from MutableMapping, I *think* that bool(mapper) is essentially a test for emptiness, like bool(dict). The behaviour is not explicitly defined in fsspec.

**Comment by rabernat** [comment]
Ok great, so let's go forward with this.

@ocefpaf - clearly the situation in #59 was not covered by our tests. Could you manage to add a simple test for this scenario?  I believe the right combo is
- rechunking an xarray dataset
- passing a mutable mapping (rather than path string) as temp_store / target_store

**Comment by rabernat** [comment]
I would add an additional parameterization here:

https://github.com/pangeo-data/rechunker/blob/17053cc2956ab8d417a666cc8675bcb4df1c5c56/tests/test_rechunk.py#L45-L54

Which parameterizes the store arguments (rather than setting them within the test function).

File: tests/test_rechunk.py [hunk 1]
```diff
@@ -47,11 +48,27 @@ def test_invalid_executor():
...
-    temp_store = str(tmp_path / "temp.zarr")
+    if target_store.startswith("mapper"):
+        target_store = fsspec.get_mapper(str(tmp_path) + target_store)
+    else:
+        target_store = str(tmp_path / target_store)
+
+    if temp_store.startswith("mapper"):
+        temp_store = fsspec.get_mapper(str(tmp_path) + temp_store)
+    else:
+        temp_store = str(tmp_path / temp_store)
```
Comment (ocefpaf) [inline tests/test_rechunk.py:71]: Not the most intelligent way to do this but I'll iterate on it later if I get better ideas.
Passing the `fsspec.get_mapper(target_store)` in the `parametrize` decorator would be worse b/c then we need to clean up the tmp files ourselves and `tmp_path` only exists when the function is execute.

TL;DR I'm avoid writing a mapper fixture here b/c I'm lazy :smile:

**Comment by rabernat** [comment]
Thanks Felipe! This is a fine solution.
